### PR TITLE
Removed snapshots and refactored tests with testing-library

### PR DESCRIPTION
**Pull request by chrisvanmook**
## Amsterdam styled components pull request

Before opening a pull request, please ensure:

- [ ] You've added or updated the README.md of the component
- [ ] You have been following the guidelines, written in the [README](../blob/master/docs/CONTRIBUTING.md#user-content-conventions-and-rules) file
- [ ] Your code has the necessary tests written
- [ ] You've exported the component in [index.ts](./packages/asc-ui/src/index.ts)
- [ ] You have updated the [CHANGELOG.md unreleased sections](../blob/master/CHANGELOG.md#muser-content-unreleased)

Be kind to code reviewers, please try to keep pull requests as small and focused as possible :)

### CODE REVIEW

File: packages/asc-ui/src/components/Toggle/Toggle.test.tsx [hunk 1]
```diff
@@ -1,28 +1,30 @@
...
+    expect(container.querySelector('button')).toBeDefined()
   })
 
   it('should display the children when the button is clicked', () => {
-    expect(component.find('#child').exists()).toBe(false)
-    component.find(ToggleButton).at(0).simulate('click')
-    expect(component.find('#child').exists()).toBe(true)
+    expect(queryByTestId('child')).toBeNull()
+    fireEvent.click(button)
+    expect(queryByTestId('child')).toBeDefined()
```
Comment (aditudorache) [inline packages/asc-ui/src/components/Toggle/Toggle.test.tsx:27]: For testing DOM elements I would use toBeInTheDocument instead or toBeDefined (https://medium.com/better-programming/how-to-do-the-basics-in-react-testing-library-c52cdb3a7726)

**Comment by robinpiets** [comment]
Hi @chrisvanmook, just did `hygen component new` and it created a `.test.tsx. file which included `expect(component).toMatchSnapshot()`. So better change the file `amsterdam-styled-components/_templates/component/new/test.ejs.t` as well

File: packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx [hunk 2]
```diff
@@ -17,10 +17,6 @@ describe('Checkbox', () => {
     ))
   })
 
-  it('should render', () => {
```
Comment (jonkoops) [inline packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx:20]: I am not seeing a replacement test for this one.

File: packages/asc-ui/src/components/DocumentCover/DocumentCover.test.tsx [hunk 3]
```diff
@@ -1,34 +1,23 @@
...
-import Button from '../Button'
+import ThemeProvider from '../../theme/ThemeProvider'
 
 jest.useFakeTimers()
 
 describe('DocumentCover', () => {
-  const mockFn = jest.fn()
-  let component: any
-
-  it('should render', () => {
```
Comment (jonkoops) [inline packages/asc-ui/src/components/DocumentCover/DocumentCover.test.tsx:12]: No replacement test for this one?

File: packages/asc-ui/src/components/GlobalStyle/GlobalStyle.test.tsx [hunk 4]
```diff
@@ -1,25 +1,13 @@
...
 import GlobalStyle from './GlobalStyle'
-import { ThemeProvider, ascDefaultTheme } from '../../theme'
+import { ThemeProvider } from '../../theme'
 
 const overrides = {
   globalStyle: `p {margin-bottom:8px;}`,
 }
 
 describe('GlobalStyle', () => {
-  it('should apply the the global style from theme', () => {
```
Comment (jonkoops) [inline packages/asc-ui/src/components/GlobalStyle/GlobalStyle.test.tsx:11]: Why was this test removed?

File: packages/asc-ui/src/components/GlobalStyle/GlobalStyle.ts [hunk 5]
```diff
@@ -8,8 +8,6 @@ const GlobalStyle = createGlobalStyle`
 
   body {
     font-family: ${getValueFromTheme('typography.fontFamily')};
-    -webkit-font-smoothing: antialiased;
-    -moz-osx-font-smoothing: grayscale;
```
Comment (jonkoops) [inline packages/asc-ui/src/components/GlobalStyle/GlobalStyle.ts:12]: I think this was from a merge conflict that was perhaps resolved incorrectly? I don't think we want to remove this.

File: packages/asc-ui/src/components/Header/HeaderLogoTextStyle.ts [hunk 6]
```diff
@@ -1,10 +1,6 @@
 import styled from 'styled-components'
 
-type Props = {
```
Comment (jonkoops) [inline packages/asc-ui/src/components/Header/HeaderLogoTextStyle.ts:3]: We're no longer offering the `tall`  version of the header? Why is that?

File: packages/asc-ui/src/components/Menu/MenuToggle/MenuToggle.test.tsx [hunk 7]
```diff
@@ -1,49 +1,10 @@
...
   it('should render', () => {
-    expect(component.exists()).toBeTruthy()
-  })
-
-  it('should render the children', () => {
-    expect(component.find(MenuList).exists()).toBeTruthy()
-    expect(component.find(MenuList).props()).toEqual({ children })
-  })
-
-  it('should toggle onClick', () => {
```
Comment (jonkoops) [inline packages/asc-ui/src/components/Menu/MenuToggle/MenuToggle.test.tsx:32]: Can't we test the visibility of the element?

File: packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx (hunk 2 above)
Comment (chrisvanmook) [reply packages/asc-ui/src/components/Checkbox/Checkbox.test.tsx:20]: Well the other tests in this spec already cover that, so I think it's pointless to add one

File: packages/asc-ui/src/components/DocumentCover/DocumentCover.test.tsx (hunk 3 above)
Comment (chrisvanmook) [reply packages/asc-ui/src/components/DocumentCover/DocumentCover.test.tsx:12]: same

File: packages/asc-ui/src/components/Header/HeaderLogoTextStyle.ts (hunk 6 above)
Comment (chrisvanmook) [reply packages/asc-ui/src/components/Header/HeaderLogoTextStyle.ts:3]: because the prop isn't used in this component :)

**Pull request merged**